### PR TITLE
changes for enabling service account to access apis

### PR DESCRIPTION
**Pull request by saravankumarpa**
- Service accounts with correct corp type can access any api under for that corp type.

- introduced corp type column in the entity table and associated changes

- Fix for email template wrong variable

*Issue #:*
https://github.com/bcgov/entity/issues/2083

*Description of changes:*

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/models/corp_type.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from .base_model import BaseCodeModel
+
+
+class CorpType(BaseCodeModel):  # pylint: disable=too-few-public-methods # Temporarily disable until methods defined
+    """This class manages all of the base data about a Corp Type."""
+
+    __tablename__ = 'corp_type'
+
+    @classmethod
+    def get_default_status(cls):
```
Comment (pksumesh) [inline auth-api/src/auth_api/models/corp_type.py]: function name is misleading, get_default_type or corp_type would be better I think.

File: auth-api/src/auth_api/models/entity.py [hunk 2]
```diff
@@ -37,8 +38,10 @@ class Entity(BaseModel):  # pylint: disable=too-few-public-methods
     pass_code_claimed = Column('pass_code_claimed', Boolean(), default=False)
     business_number = Column('business_number', String(100), nullable=True)
     name = Column('name', String(250), nullable=True)
+    corp_type_code = Column(String(10), ForeignKey('corp_type.code'), nullable=False)
 
     contacts = relationship('ContactLink', back_populates='entity')
+    corp_type = relationship(CorpType, foreign_keys=[corp_type_code], lazy='joined', innerjoin=True)
```
Comment (pksumesh) [inline auth-api/src/auth_api/models/entity.py]: A lazy load relationship would match with our existing pattern.

File: auth-api/src/auth_api/services/authorization.py [hunk 3]
```diff
@@ -49,6 +49,15 @@ def get_user_authorizations_for_entity(token_info: Dict, business_identifier: st
             auth_response = {
                 'roles': ['edit', 'view']
             }
+        elif Role.SYSTEM.value in token_info.get('realm_access').get('roles'):
+            # a service account in keycloak should have CORP_TYPE claim setup.
+            keycloak_corp_type = token_info.get('CORP_TYPE', None)
```
Comment (pksumesh) [inline auth-api/src/auth_api/services/authorization.py]: claim name needs to be in small case.

Comment (saravankumarpa) [reply auth-api/src/auth_api/services/authorization.py]: done

File: auth-api/src/auth_api/models/entity.py (hunk 2 above)
Comment (saravankumarpa) [reply auth-api/src/auth_api/models/entity.py]: done

**Comment by saravankumarpa** [comment]
image
image

**Pull request merged**